### ISSUE TITLE
How do I migrate from 2.0?

### ISSUE THREAD

**Issue by kykim** [octopress/octopress#29]
Subject says it all. Is there a doc? I can't seem to find it (but then again I've only had one coffee so far.)

Thanks,
-kevin

**Comment by parkr** [comment]
Octopress 3.0 has not yet been released.

**Comment by kelcecil** [comment]
I just finished this process with my simple Octopress blog, and it was pretty straight forward. I'll share what I learned.

I installed Octopress 3 and Jekyll and created a blog in a new folder with `octopress new <dir>`. I copied my posts directly from the Octopress 2 _posts folder to the new _posts folder. I only used codeblock tags and the default theme, so my only necessary modification was to replace `{% codeblock %}...{% endcodeblock %}` to `{% highlight %}...{% endhighlight %}`.

I deploy my site by including the files inside of a Docker container, so I use jekyll build to generate my website to add to the container. This places everything inside the _site folder. The only gotcha I ran into was that jekyll will put posts into folders named after their categories (eg. http://kelcecil.com/golang/2015/01/09/using-go-generate-in-go-1-dot-4.html). This may break links, so be careful.

If you run into errors regarding failing to read a header, then upgrading Python appears to take care of it (https://github.com/jekyll/jekyll/issues/1181). You might see an error regarding a duplicate key named index0 when building your site; it's a problem in Liquid 2.6.1, but it's been recently addressed (https://github.com/Shopify/liquid/commit/5df1a262ad898918ad382a4d87c42e831db6babb). Your site should generate fine regardless.

Hope this helps!

**Comment by imathis** [comment]
Thanks for sharing @kelcecil!

**Comment by Mazyod** [comment]
@kelcecil Cool stuff! Would love to see some expansion regarding migrating themes, else I'll try it someday.

**Comment by FranklinChen** [comment]
I got excited by today's JekyllConf (just watched a bunch of the YouTube video), and hope that an official migration guide will be provided from Octopress 2 to 3.

**Comment by imathis** [comment]
Yeah, I think I can make a new guide, but it won't involve a theme yet since I haven't finished the new default theme. You can actually start using the Octopress CLI for creating posts and deploying now without changing anything.

**Comment by AnthonyMastrean** [comment]
I was just looking for this information. I'll try something like @kelcecil's instructions and let you know how it goes.

**Comment by plttn** [comment]
One point of note, if you're used to using pow to get a .dev address rather than using localhost:4000, it won't work so well without some fudging either on the jekyll side or the pow side. 

Fixing it on the site side requires you to either
1. use `public` instead of `_site`, like legacy octopress
2. use `rack-jekyll` (which is kinda outdated, and in my testing, not as fast as it could be)

Fixing it on pow side requires you to do a bit more work with the symlink

```
$ cd ~/.pow
$ mkdir octo3
$ cd octo3
$ ln -s ~/Github/octo3 public
```

**Comment by fgimian** [comment]
I might give this migration a shot too folks.  Based on the releases, am I correct in guessing that Octopress 3.0 is now considered stable?

**Comment by jslim89** [comment]
@kelcecil Is it possible to fix the post URL with **blog** prefix? e.g. http://kelcecil.com/golang/2015/01/09/using-go-generate-in-go-1-dot-4.html, fix it to http://kelcecil.com/blog/2015/01/09/using-go-generate-in-go-1-dot-4.html instead?

**Comment by phatblat** [comment]
Check out: http://samwize.com/2015/09/30/migrating-octopress-2-to-octopress-3

**Comment by dgmstuart** [comment]
I've just completed a migration - I've taken a slightly different approach to some of the other posts and blogged about it here: https://dgmstuart.github.io/2016/01/22/migrating-from-octopress-2-to-3.html

**Comment by dentex** [comment]
Wonderful!
I read the blog and I'll try the migration ASAP!

Regards.
Il 22/gen/2016 02:05, "Duncan Stuart" <notifications@github.com> ha scritto:

**Comment by dgmstuart** [comment]
@dentex in the cold light of day it is a bit more complicated: Octopress 2 provided a full blogging service with hooks for comments, analytics, themes etc. Octopress 3 is now more like a set of tools for generating content for a Jekyll blog, so it looks like all that sort of functionality belongs in the Jekyll theme.

I think a better approach might be:
1. Pick a Jekyll theme with the features you want and set up a site using that theme.
2. Use `octopress init` to add the page and post templates
3. Import data in the same way

I feel like a new blog post is in order.

**Comment by dentex** [comment]
Well OK.
Thanks for the additional info.
Actually I can't try anything for a couple of days.  :)
Il 22/gen/2016 19:46, "Duncan Stuart" <notifications@github.com> ha scritto:

**Comment by ishu3101** [comment]
How do you migrate the default octopress 2 theme to octopress 3? Octopress 2 uses [classic](https://github.com/octopress/classic-theme) as the default theme while octopress 3 uses the default Jekyll theme.

**Comment by dentex** [comment]
More than a year has passed, and every time I try to migrate it's a mess... Too many missing pieces for me.

**Comment by awongCM** [comment]
Hello all,
Has anybody else figure out how to successfully upgrade/migrate your Octopress 2.0 to Octopress 3.0 in the Heroku environment?
I'm just wondering has anybody gone through this route before and be willing to share some experience how to go about it?

**Comment by dgmstuart** [comment]
@awongCM see my comments above. In my opinion one shouldn’t think of it as an upgrade because Octopress 3 isn’t a new version of Octopress 2 - they’re completely different things: Octopress 2 was more or less a stand-alone blogging platform, while Octopress 3 is a set of tools for managing Jekyll blogs. 

So instead I’d think of it as porting your Octopress blog over to Jekyll...

**Comment by awongCM** [comment]
@dgmstuart Thanks for that.

So, from your point of view, does Octopress 3 come with any major performance boost benefits?

I noticed that on my Octopress 2 blog, every time I run `rake preview`, and then write some changes to one of my blog content post markdown, the `rake watch` process will take considerable time to regenerate the site (~average about 10-15 secs).  I feel this is very slow when getting content review and feedback as quickly as I possibly can before publishing.  

Even tiny changes such as a text change also take the same length of time to render.

Thus I am hoping Octopress 3 can help in resolving this slowness issue...

**Comment by dgmstuart** [comment]
@awongCM not sure, sorry. I think one of the main improvements was that maintenance of Octopress2 was painful, since it’s more like a fork of someone else’s blog, rather than a library or platform. But as far as I know both Octopress 2 and 3 haven’t been maintained for a couple of years, so that’s not really a thing

**Comment by awongCM** [comment]
@dgmstuart Okay.  That's a bummer.

If that's the case, I see no benefit gain from doing this "upgrade" work at all.

I might as well consider replacing Octopress 2 with another more powerful/faster blogging platform for tech writers like myself should Octopress 2 gets too 'shaky' to handle future blog content I'll be posting.

Thank you so much for your time anyways. Much appreciated!
